evalml/model_understanding/prediction_explanations/_user_interface.py

```diff
@@ -423,3 +451,25 @@ def make_dict(self, data):
                                                                  data.input_features)["explanations"]
             report.append(section)
         return {"explanations": report}
+
+    def make_dataframe(self, data):
```

Comment:
Is there some way we can have this implementation use `_make_single_prediction_shap_table` to make the tables per-prediction, and then concatenate them together? This would avoid confusion and bugs stemming from current or future differences in the implementations.

Reply:
I think that's what we're doing! `self.table_maker.make_dataframe(index, data.pipeline, data.input_features)` calls `_make_single_prediction_shap_table` to make the shap table per prediction. Then we add metadata to each table (prediction probabilities, etc) and then we concat all the tables at the end.